File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
        int[] freqs = new int[26];
        for (char c : word.toCharArray()) {
            freqs[c - 'a']++;
        }
        
        Arrays.sort(freqs);
        // System.out.println(Arrays.toString(freqs));
        int pushes = 0;
        for (int i = 25; i >= 0; i--) {
            pushes += freqs[i] * (((25 - i) / 8) + 1);
        }

        return pushes;
    }
}

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        pushes = 0
        freqs = sorted(collections.Counter(word).values(), reverse=True)

        # print(freqs)
        for i, f in enumerate(freqs):
            multiple = i // 8 + 1
            pushes += f * multiple

        return pushes
